src/rws_poco_client.cpp

```diff
@@ -334,8 +334,19 @@ POCOClient::POCOResult POCOClient::webSocketConnect(const std::string uri,
   try
   {
     result.addHTTPRequestInfo(request);
-    p_websocket_ = new WebSocket(http_client_session_, request, response);
-    p_websocket_->setReceiveTimeout(Poco::Timespan(timeout));
+    {
+      // We must have atleast websocket_connect_mutext_,
+      // but if a connection already exists, we must also have websocket_use_mutex_.
+      ScopedLock<Mutex> connect_lock(websocket_connect_mutex_);
+      if (webSocketExist()) {
```

Comment:
```suggestion
      if (webSocketExist())
      {
```
Curly bracket on new line to be consistent with the rest of the code base.